client/modules/Booking/components/TimeTable.js

```diff
@@ -17,102 +18,215 @@ import {
 
+//
+const days = ["Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"];
+const hours = [
+  "00:00",
+  "01:00",
+  "02:00",
+  "03:00",
+  "04:00",
+  "05:00",
+  "06:00",
+  "07:00",
+  "08:00",
+  "09:00",
+  "10:00",
+  "11:00",
+  "12:00",
+  "13:00",
+  "14:00",
+  "15:00",
+  "16:00",
+  "17:00",
+  "18:00",
+  "19:00",
+  "20:00",
+  "21:00",
+  "22:00",
+  "23:00",
+];
 
 //Cars component class
 export class TimeTable extends Component {
+  sun = moment().startOf("week");
+  mon = moment()
+    .startOf("week")
+    .add(1, "d");
+  tue = moment()
+    .startOf("week")
+    .add(2, "d");
+  wed = moment()
+    .startOf("week")
+    .add(3, "d");
+  thu = moment()
+    .startOf("week")
+    .add(4, "d");
+  fri = moment()
+    .startOf("week")
+    .add(5, "d");
+  sat = moment()
+    .startOf("week")
+	.add(6, "d");
+  selected= moment()
+  .startOf("hour")
+  .add(1, "hours");
+	
 
-	constructor(props){
-		super(props);
-		this.weekBackward = this.weekBackward.bind(this);
-		this.weekForward = this.weekForward.bind(this);
-		this.state = {
-			times:[],
-			time:moment().startOf('week'),
-		}
-	}
+  constructor(props) {
+    super(props);
+    this.weekBackward = this.weekBackward.bind(this);
+    this.weekForward = this.weekForward.bind(this);
+    this.state = {
+      times: [],
+      time: moment().startOf("week"),
+      days: [
+        this.sun,
+        this.mon,
+        this.tue,
+        this.wed,
+        this.thu,
+        this.fri,
+        this.sat,
+      ],
+    };
+  }
 
-	componentDidMount(){
-		this.loadData(this.state.time);
-	}
-	
-	loadData(time){
-		http.client().get(`/cars/${this.props.data._id}/times?start=${time.toISOString()}`).then(res => {
-			this.setState({
-				times: res.data,
-			})
-		})
-	}
+  componentDidMount() {
+    this.loadData(this.state.time);
+    this.props.onRef(this);
+  }
+
+  componentWillUnmount() {
+    this.props.onRef(undefined);
+  }
+
+  loadData(time) {
+    http
+      .client()
+      .get(`/cars/${this.props.data._id}/times?start=${time.toISOString()}`)
+      .then(res => {
+        this.setState({
+          times: res.data,
+        });
+      });
+  }
+
+  /*
+  Attempt at callback function
+  Appears to be called but can not change value
+  */
+
+  onItemClickHandler(itemName) {
+	this.props.func(itemName);
+  }
 
-	renderHour(h){
-		return(
-			<tr key={h}>
-				<th scope="row">{h}</th>
-				{days.map(d => <td key={d+h}>{this.checkBooking(d, h) ?  " " : "✓"}</td>)}
-			</tr>
-		)
-	}
+  renderHour(h) {
+    return (
+      <tr key={h}>
+        <th scope="row">{h}</th>
+        {this.state.days.map(d => {
+    var clickHandler = (event => {
+        return event => {
+          this.onItemClickHandler(moment(d).add(h.substring(0,2),'h'));
+        };
+      })(d);
+   return (
+         <td key={d + h}>
+             {this.checkBooking(moment(d).format("ddd"), h) ? " " : 
+              <Button onClick={clickHandler}>✓</Button>}
+         </td>
+       );
+	 }
+)}
```

Comment:
but overall, it's not necessarily a good idea to go this deep into nested loops here, and do this sort of closure trickery to get your values into the click handler. It might be much easier for you (and much more readable) to set the value as a data attribute on Button, and then read it back in the handler via `event.target` accessor - I believe it will greatly simplify this code